tio/bufferedreader.go

```diff
@@ -257,6 +272,55 @@ func (buffer *BufferedReader) parseMLE64() ([]byte, int) {
 	return buffer.extractMessage(int(messageLen), buffer.paramMLE+8)
 }
 
+// messages are separated by regexp
+func (buffer *BufferedReader) parseDelimiterMultiLine() ([]byte, int) {
+	startRegexp := regexp.MustCompile(buffer.delimiterStr)
+	lineDelimiter := []byte("\n")
+
+	hasMessage := false
+	messageStarted := false
+	loopStartIdx := 0
+	endIdx := 0
+
+	for true {
+		endIdx = bytes.Index(buffer.data[loopStartIdx:buffer.end], lineDelimiter)
```

Comment:
golang regexp has multiline support.
See https://github.com/google/re2/wiki/Syntax and https://github.com/trivago/gollum/issues/228

Reply:
I have tried multiline regexp. but in case of using regexp last message did not get captured until next message arrives. 
this implementation will capture last message